scala-package/core/src/main/scala/ml/dmlc/mxnet/module/Module.scala

```diff
@@ -314,7 +316,10 @@ class Module(symbolVar: Symbol,
         }
 
       optimizerInitialized = true
-      // TODO: preload optimizer states
+      preloadOptStates.foreach { optStates =>
+        loadOptimizerStates(optStates)
+        preloadOptStates = None
```

Comment:
`preloadOptStates` is set when Module.loadCheckpoint is called, it should be reset in case users choose to use another optimizer.